src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -364,7 +364,8 @@ private static IUnboundLambdaExpression CreateUnboundLambdaOperation(UnboundLamb
             // an IUnboundLambdaExpression, you need to look at the parent IConversionExpression.
             ITypeSymbol type = null;
             Optional<object> constantValue = ConvertToOptional(unboundLambda.ConstantValue);
-            return new UnboundLambdaExpression(isInvalid, syntax, type, constantValue);
+            Lazy<ILambdaExpression> internalLambda = new Lazy<ILambdaExpression>(() => CreateBoundLambdaOperation(unboundLambda.BindForErrorRecovery()));
```

Comment:
Given the other conversation thread for tests (about identity of the node), it feels like we should simply request the IOperation node for the lambda syntax from the SemanticModel. We would have to ensure that SemanticModel would never "give" an unbound lambda for a request like that and would do appropriate caching.

---
In reply to: [121856689](https://github.com/dotnet/roslyn/pull/20197#discussion_r121856689) [](ancestors = 121856689)